[QuekBingHeng] iP

QuekBingHeng opened the pull request:
# Angel Chatbot Project 🚀

Welcome to the **Angel Chatbot** project! This chatbot is designed to help you manage your tasks with ease. It supports various commands to find, add, mark, and delete tasks, making your life more organized.

## Features

- Supports various task types: `todo`, `deadline`, `event`, and 'find'.
- Easily mark tasks as done or undone.
- Search for tasks using keywords.
- Saves tasks automatically to a file for persistence.
- User-friendly command parsing with detailed feedback.

## How to Use

1. Clone the repository:
   ```bash
   git clone https://github.com/username/angel-chatbot.git

2. Navitgate to the project directory:
   ```bash
   cd angel-chatbot

3. Compile the program:
   ```bash
   javac -d bin src/main/java/angel/*.java

4. Run the application:
   ```bash
   java -cp bin main.java.angel.Angel

## Task List

- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events

## Additional Resources
For more information, checkout the [Github Documentation](https://docs.github.com/en).
> "Great things are not done by impulse, but by a series of small things brought together." — Vincent Van Gogh

## 
If you are a Java Programmer you can use it to practice code too. Heres the 'main' method:
```bash
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

QuekBingHeng committed:
add Level-0, Angel.java

QuekBingHeng committed:
add Level-1, Angel.java

QuekBingHeng committed:
add Level-2, Angel.java

QuekBingHeng committed:
add Level-3, Angel.java

QuekBingHeng committed:
add Level-4, Angel.java

QuekBingHeng committed:
add Level-5, Angel.java

QuekBingHeng committed:
add Level-6, Angel.java

QuekBingHeng committed:
add A-TextUiTesting

QuekBingHeng committed:
Add Level-7, Angel.java with classes

QuekBingHeng committed:
Add Level-8, Angel.java

QuekBingHeng committed:
Add TaskList.java

QuekBingHeng committed:
Add Ui.java

QuekBingHeng committed:
Add Parser.java

QuekBingHeng committed:
Add package main.java.angel

QuekBingHeng committed:
Add Angel.txt which is the chatbot list

QuekBingHeng committed:
Merge branch 'add-gradle-support'

QuekBingHeng committed:
Add DeadlineTest.java and EventTest.java

QuekBingHeng committed:
Add .jar file

QuekBingHeng committed:
Add JavaDoc comments

QuekBingHeng committed:
Tweak code to comply with coding standard

QuekBingHeng committed:
Add Level-9, Angel.java

thenosewizard commented on src/main/java/angel/Parser.java:
```diff
@@ -0,0 +1,265 @@
+    }
+
+    /**
+     * Parses the command for creating a deadline task.
+     *
+     * @param parts The parts of the user input split by space.
+     * @return A Command object for the deadline task.
+     * @throws InvalidCommandException If the deadline description or date/time is missing or incorrectly formatted.
+     */
```
I like how you have framed most of your javaDoc descriptions. Simple and easy to read

thenosewizard commented on src/main/java/angel/Parser.java:
```diff
@@ -0,0 +1,265 @@
+            return type;
+        }
+
+        /**
+         * Gets the details of the command.
+         *
+         * @return The command details.
+         */
```
Perhaps the object type that is returned could be specified in the description of the function.

QuekBingHeng replied:
Okay got it!

thenosewizard reviewed (commented):
LGTM :thumbsup:. Coding standards are followed thoroughly.

QuekBingHeng committed:
Update README.md

QuekBingHeng committed:
Update README.md

QuekBingHeng committed:
Update README.md

QuekBingHeng committed:
Update README.md

QuekBingHeng committed:
Update README.md

marraknas commented on src/main/java/angel/Angel.java:
```diff
@@ -0,0 +1,106 @@
+                Parser.Command command = Parser.parseCommand(input);
+
+                switch (command.getType()) {
+                case "bye":
```
I think maybe you can refactor all these code into one class called "Handler" which handles all these inputs!

QuekBingHeng replied:
I will try! Thanks!

marraknas commented on src/main/java/angel/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+    @Override
+    public String toString() {
+        // Format the due date/time string with a comma
+        return "[D][" + getStatusIcon() + "] " + description
```
Good that you follow coding standards when splitting between 2 lines!

marraknas commented on src/main/java/angel/Parser.java:
```diff
@@ -0,0 +1,265 @@
+package main.java.angel;
+
+import java.time.LocalDateTime;
```
very small problem but I think you can put the import statements in lexicographical order!

QuekBingHeng replied:
Done!

marraknas commented on src/main/java/angel/Parser.java:
```diff
@@ -0,0 +1,265 @@
+        case "delete":
+            return parseIndexCommand(parts, command);
+        default:
+            throw new InvalidCommandException("Unknown command: " + command + ". Please enter a valid command (e.g., list, todo, deadline, event, mark, unmark, delete, bye, find).");
```
quite a long line, maybe you can split it into two lines according to the coding standards!

marraknas commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,66 @@
+/**
+ * Test class for the Deadline class.
+ */
+class DeadlineTest {
```
comprehensive tests! Good 😊

marraknas reviewed (commented):
lgtm! Very few coding errors and just a few nits to fix. Otherwise, well written code and can do a bit of refactoring as mentioned! Nice 👍

QuekBingHeng committed:
Add Checkstyle

QuekBingHeng committed:
Add README

QuekBingHeng committed:
Add README

QuekBingHeng committed:
Add DialogBox.java, GUI.java, Launcher.java, at JavaFX tutorial part 3

QuekBingHeng committed:
Add Level-10, DialogBox.fxml, GuiWindow.fxml, GuiWindow.java, add FXML changes to DialogBox.java, Gui.java, and rename GUI.java into Gui.java

QuekBingHeng committed:
Update README.md

QuekBingHeng committed:
Add assertions to GuiWindow.java

Assertions have been added to the GuiWindow.java class to ensure that
critical components such as scrollPane, dialogContainer, and angel are
correctly initialised. This helps catch potential runtime errors early
and improves overall code robustness.

The handleUserInput method now includes assertions to validate that
user inputs are non-null and non-empty before processing, ensuring that
only valid data is handled by the chatbot. Similarly, the response
from Angel is asserted to be non-null, providing additional safeguards.

These changes enhance error detection during development and make the
code more maintainable by explicitly stating assumptions about
component states and input validity.

QuekBingHeng committed:
Add assertions to Launcher.java

Added assertions in Launcher.java to validate command-line arguments
and class references before launching the application. These assertions
ensure that key elements are not null, enhancing the robustness of the
main method and helping to catch potential issues early during runtime.

While assertions in this class are minimal, they serve as a basic
safeguard and demonstrate good practices for future extensions.

QuekBingHeng committed:
Add assertions to Gui.java

Enhanced the robustness of the Gui.java by adding assertions
throughout the start() method. These assertions ensure critical
conditions are met during the initialisation and display of the
JavaFX components.

Key changes include:
- Added an assertion to check if the Stage object is not null
  before proceeding with the initialisation.
- Verified that the FXMLLoader and AnchorPane are properly loaded
  and not null, preventing potential issues with missing FXML files.
- Added an assertion to check that the GuiWindow controller is
  correctly set before injecting the Angel instance.
- Ensured that the stage is visible after calling stage.show(),
  adding an extra layer of verification to the UI setup.
- Handled exceptions with an assertion fail to clearly indicate
  when an IOException disrupts the loading process.

These improvements provide better error reporting and help identify
potential issues early in the application lifecycle, improving
overall code quality and maintainability.

QuekBingHeng committed:
Add assertions to DialogBox.java

QuekBingHeng committed:
Merge pull request #1 from QuekBingHeng/branch-A-Assertions

Comparing A-Assertions to master

QuekBingHeng committed:
Merge pull request #2 from QuekBingHeng/master

Merging master to A-CodeQuality

QuekBingHeng committed:
Add CodeQuality to Gui.java

QuekBingHeng committed:
Merge pull request #3 from QuekBingHeng/branch-A-CodeQuality

Merge CodeQuality to master

QuekBingHeng committed:
Add CodeQuality changes from feedback

QuekBingHeng committed:
Add C-DetectDuplicates

QuekBingHeng committed:
Add CheckStyle and CodeQuality feedback

QuekBingHeng committed:
Add A-UserGuide

QuekBingHeng committed:
Add A-Release

QuekBingHeng committed:
Add iP user guide render solution
